### DISCUSSION TITLE
How to auto redirect to provider in authjs v5

### DISCUSSION THREAD

**Discussion by ryub3n** [nextauthjs/next-auth#10886] [Help]
I have a single provider Microsoft Entra ID (Azure AD) and I want the user to be auto redirected to the sign-in page of Microsoft Entra ID when he vists the site as below:

- User logins to the website, ie. localhost:3000/
- If the user is not logged in, the user is redirected to the Microsoft Entra ID login page
- Upon successful login, the user is redirected to the homepage.
- If the user logs out, the user is again redirected to the Microsoft Entra ID login page

How can this be done with Next-Auth v5 and NextJS 14 app directory?

Ref: https://github.com/nextauthjs/next-auth/discussions/4078

**Comment by finleysg** [comment]
There is a simple solution in the [thread you referenced above](https://github.com/nextauthjs/next-auth/discussions/4078#discussioncomment-9806999) that works for Entra ID.
